Add ir frontend doc (#13351)
openvinotoolkit/openvino: src/frontends/ir/docs/architecture.md

# OpenVINO IR Frontend Architecture

> OpenVINO IR Frontend uses [pugixml](../../../thirdparty/pugixml/README.md) library to parse xml files. After that based on version and name of operation we found supported operation and create it using OpenVINO Visitor API:

Reviewer: Could you please clarify who is we in the second sentence? Sounds a bit confusing

> OpenVINO IR Frontend uses [pugixml](../../../thirdparty/pugixml/README.md) library to parse xml files. After that based on version and name of operation frontend found supported operation and creates it using OpenVINO Visitor API:

Reviewer: ```suggestion
OpenVINO IR Frontend uses [pugixml](../../../thirdparty/pugixml/README.md) library to parse XML files. After that, based on the version and name of the operation, the Frontend creates the supported operation and initializes it using OpenVINO Visitor API:
```

```mermaid
flowchart TB
    fw_model[(IR)]
    style fw_model fill:#427cb0
    
    pugixml([pugixml])
    subgraph frontend [ov::frontend::ir::FrontEnd]
        load_impl["load_impl()"]
    end
    fw_model--as stream-->load_impl
    load_impl--load stream-->pugixml
    pugixml--parsed object-->load_impl
    
    
    subgraph input_model [ov::frontend::ir::InputModel]
        convert["convert()"]
    end
    
    load_impl--create-->input_model
    
    xml_deserializer(ov::XmlDeserializer)
    ov_model[ov::Model]
    
    convert--create visitor-->xml_deserializer
    
    xml_deserializer--recursively parse all operations from the model-->ov_model
```

## Extensions

> OpenVINO IR Frontend supports extensions in order to add some extension you can use `ov::frontend::ir::Frontend::add_extension()` API.

Reviewer: ```suggestion
OpenVINO IR Frontend supports extensions. To add an extension, use `ov::frontend::ir::Frontend::add_extension()` API.
```

The next extension types are supported:

> * `ov::TelemetryExtension` - enable telemenry for frontend
> * `ov::BaseOpExtension` - enable support of custom operation
> * `ov::detail::SOExtension` - allow to support `ov::BaseOpExtension` extensions which were loaded from the external library.

Reviewer: ```suggestion
* `ov::TelemetryExtension` - enable telemetry for the frontend
* `ov::BaseOpExtension` - enable support of a custom operation
* `ov::detail::SOExtension` - allow to support `ov::BaseOpExtension` extensions loaded from the external library.
```
